File: crates/ariadne-parser/fixtures/java/Sample.java
package com.ariadne.sample;

import java.util.ArrayList;
import java.util.List;

public class Sample {
    public interface Step {
        int next(int current);
    }

    public enum Side {
        LEFT,
        RIGHT,
    }

    public record Point(int x, int y) {}

    public static int increment(int v) {
        List<Integer> values = new ArrayList<>();
        values.add(v + 1);
        return values.get(0);
    }
}

class Helper {
    @Deprecated
    public void legacy() {}

    @Override
    public String toString() {
        return "helper";
    }
}

File: crates/ariadne-parser/fixtures/csharp/Sample.cs
using System;
using System.Collections.Generic;

namespace Ariadne.Sample
{
    public interface IStep
    {
        int Next(int current);
    }

    public enum Side
    {
        Left,
        Right,
    }

    public struct Point
    {
        public int X;
        public int Y;
    }

    public record Vec(int X, int Y);

    public class Counter
    {
        public int Value;

        public int Increment()
        {
            Value += 1;
            return Value;
        }

        public static int Tick(int v)
        {
            var list = new List<int>();
            list.Add(v);
            return Console.In.Read();
        }
    }
}
